app/controllers/scraper_results_controller.rb

```diff
@@ -3,7 +3,14 @@ def index
     query = params[:query]
     scraper = current_user.labels.find(current_label_ids).map(&:default_scraper).compact.first
     results = scraper.new(query: query).scrape
-    results = FuzzySubstringMatchFilter.new(results, query).matches
+
+    relevant_results = results.select { |result| result[:is_reliable] }
+    matching_results = FuzzySubstringMatchFilter.new(results, query).matches
+    
+    # intersection of relevant and matching results
+    results = relevant_results & matching_results
+    results = relevant_results if results.empty?
```

Comment:
If there was no intersection between the relevant- and matching results, then we return only the relevant results. this happens for example in the james bond example: the word "james bond" is not contained in the title (e.g. the movie is called _skyfall_ and not _james bond skyfall_). Still, irrelevant results that have the same title as a famous title will be filtered.